frontend/src/lib/overlay/index.ts

```diff
@@ -98,13 +110,6 @@ function buildSharedClickOverlay(): { overlayTarget: OverlayAction; overlayConta
       });
```

Comment:
this comment is for the querySelector but I can't comment on it 🤷
This will break if menu items are dynamically added or removed from the content element as we are explicitly listening for clicks on each item, instead we should probably just listen for clicks on the content element and filter based on what was clicked. Not sure how hard that is.